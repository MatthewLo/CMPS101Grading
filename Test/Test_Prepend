https://github.com/ShobhitMaheshwari/CMPS101Grading/blob/master/Test/Test_Prepend.c
5
Check list prepend
gcc -std=c99 -Wall -c Test_Prepend.c > w 2>&1
gcc -std=c99 -Wall -o Test_Prepend Test_Prepend.o ../List.o &> r
./Test_Prepend
valgrind --leak-check=full --log-file="memoryLeak" ./Test_Prepend
