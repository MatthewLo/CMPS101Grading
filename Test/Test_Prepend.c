
#include<stdio.h>
#include<stdlib.h>
#include "../List.h"
int main(){
	FILE* out;
	out = fopen("Test_Prepend.out", "w");
	if( out==NULL ){
		printf("Unable to write to file\n");
		exit(EXIT_FAILURE);
	}
	List a = newList();
	append(a, 3);
	printList(out, a);
	prepend(a, 4);
	printList(out, a);
	List b = newList();
	prepend(b, 6);
	printList(out, b);
	prepend(b, 12);
	printList(out, b);
	fclose(out);
	freeList(&b);
	freeList(&a);
	return 0;
}
