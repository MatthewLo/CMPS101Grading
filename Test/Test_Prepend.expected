3
4 3
6
12 6
